File: baekjoon/java/src/silver5/Q1789_SumOfNumbers.java
package silver5;

import java.util.Scanner;

// https://www.acmicpc.net/problem/1789
public class Q1789_SumOfNumbers {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long n = sc.nextLong();
        long start = 1;
        long end = n;
        long answer = 0;
        while(start <= end) {
            long mid = (start+end) / 2;
            if (mid * (mid+1) / 2 <= n) {
                answer = mid;
                start = mid + 1;
            } else {
                end = mid - 1;
            }
        }
        System.out.println(answer);
    }

    public static void solution(String[] args) {
        Scanner sc = new Scanner(System.in);
        long n = sc.nextLong();
        long total = 0;
        long now = 0;
        while (total < n) {
            now++;
            total += now;
        }
        if (total != n) {
            now--;
        }
        System.out.println(now);
    }
}

File: baekjoon/python/silver5/q1789_sum_of_numbers.py
## https://www.acmicpc.net/problem/1789
def solution():
    n = int(input())
    total = 0
    now = 0
    while(total<n):
        now += 1
        total += now
    if total != n :
        now -= 1
    return now

print(solution())

## 이분 탐색으로 푼 풀이를 찾아봄
def other():
    n = int(input())
    start = 1
    end = n

    while start <= end :
        mid = (start + end) // 2
        if mid * (mid+1) // 2 <= n :
            answer = mid
            start = mid + 1
        else:
            end = mid-1
    return answer

print(other)

## 근의 공식 이용해서 푼 풀이도 있다. 
